libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/NumberInput.cs

```diff
@@ -60,22 +61,21 @@ protected override Task<InputState> OnRecognizeInput(DialogContext dc)
                 return Task.FromResult(InputState.Unrecognized);
             }
 
-            dc.GetState().SetValue(VALUE_PROPERTY, input);
+            dcState.SetValue(VALUE_PROPERTY, input);
 
-            if (!string.IsNullOrEmpty(OutputFormat))
```

Comment:
>(!string.IsNullOrEmpty [](start = 15, length = 22)

There have been a couple of places where you replaced this with checking for an expression.  What happens if there is an expression that evaluates to null or empty string?

Reply:
it becomes default(T) 

---
In reply to: [367274215](https://github.com/microsoft/botbuilder-dotnet/pull/3253#discussion_r367274215) [](ancestors = 367274215)